pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IngestionDelayTracker.java

```diff
@@ -196,21 +202,28 @@ void setClock(Clock clock) {
    * Called by LLRealTimeSegmentDataManagers to post ingestion time updates to this tracker class.
    *
    * @param ingestionTimeMs ingestion time being recorded.
+   * @param creationTimeMs event creation time.
    * @param partitionGroupId partition ID for which this ingestion time is being recorded.
    */
-  public void updateIngestionDelay(long ingestionTimeMs, int partitionGroupId) {
+  public void updateIngestionDelay(long ingestionTimeMs, long creationTimeMs, int partitionGroupId) {
     // Store new measure and wipe old one for this partition
-    // TODO: see if we can install gauges after the server is ready.
     if (!_isServerReadyToServeQueries.get()) {
       // Do not update the ingestion delay metrics during server startup period
       return;
     }
-    Long previousMeasure = _partitionToIngestionTimeMsMap.put(partitionGroupId,
-        ingestionTimeMs);
+    IngestionTimestamps previousMeasure = _partitionToIngestionTimestampsMap.put(partitionGroupId,
+        new IngestionTimestamps(ingestionTimeMs, creationTimeMs));
     if (previousMeasure == null) {
       // First time we start tracking a partition we should start tracking it via metric
       _serverMetrics.setOrUpdatePartitionGauge(_metricName, partitionGroupId,
           ServerGauge.REALTIME_INGESTION_DELAY_MS, () -> getPartitionIngestionDelayMs(partitionGroupId));
+      if (creationTimeMs != Long.MIN_VALUE) {
```

Comment:
Suggest using `<=0` to be more flexible

Reply:
> 



> @jugomezv I know I missed the meeting for this PR. But can you please explain why this cannot be solved without modifying any pinot-spi?

It can but it is more work and we need to duplicate the tracker and that adds overhead on ideal state reads which we like to avoid. There are other issues like figuring when to delete the tracker etc. The current implementation is basically little overhead (extra args) and provides the path to other streams to leverage this as well

Reply:
Yes it is more work :) but I don't think you would ideal state reads overhead. other seem more minor reasoning to me. 

Anyway, I want to make sure that there is clear guidance on anyone trying to add a connector to a new source. If the guidance ensures that this field can be set optionally in the metadata and its usage, then it is fine with me.

Reply:
Well two trackers running timers thread in parallel will trigger double of reads and then we have all duplicated infrastructure, with this approach we have much less code and little or no overhead when you don't have that time stamp.

I think the new comments should make clear what this new time stamp is and if not please add some direction on what you think may be missing. Referring to commit 9bb0befe9a55c01bb3a5066a4b30137469819f38

Reply:
> Suggest using `<=0` to be more flexible

I changed to take anything larger and = 0 as valid in commit 9bb0befe9a55c01bb3a5066a4b30137469819f38 I guess zero is still a valid time if someone were to set the clock to old dates